dracut/30coreos-installer/parse-coreos.sh

```diff
@@ -2,6 +2,36 @@
 
 . /lib/dracut-lib.sh
 
+# certain kernel argument might have multiple values and
+# getarg() from dracut-lib.sh would grab the last value
+# e.g. `nameserver=192.168.122.1 namserver=1.1.1.1` would
+# only echo 1.1.1.1
+# related bug: https://bugzilla.redhat.com/show_bug.cgi?id=1763341
+getargs_multiple() {
+    local _o _val _doecho
+    local CMDLINE="$(cat /proc/cmdline)"
+    unset _val
+    unset _o
+    unset _doecho
+
+    for _o in $CMDLINE; do
+        if [ "${_o%%=*}" = "${1%%=*}" ]; then
```

Comment:
This is some `bash` sorcery!  Could you add a comment here with a little bit more detail of what this is doing?

Reply:
For sure! Should've added comments in the first place.
Also this is mostly from [this line from `/dracut/dracut-lib.sh`](https://github.com/zfsonlinux/dracut/blob/df03141dbc9896b808cebdcc7358745dc2e8b979/modules.d/99base/dracut-lib.sh#L216) but with multiple matched karg concatenated together.

Reply:
> This is some `bash` sorcery! 
It's part of POSIX shell.
https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html.